### ISSUE TITLE
Pulumi.AzureNative.Network.Inputs.SubResourceArgs.Id must not be an Input<T>

### ISSUE THREAD

**Issue by vansli** [pulumi/pulumi-azure-native#768]
I  just updated from azure-native 0.9.0 to 1.0.0 and am facing an exception during pulumi up and preview:

**[Input] Pulumi.AzureNative.Network.Inputs.SubResourceArgs.Id must not be an Input<T>**

error: Running program '.........Network.dll' failed with an unhandled exception:
    System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
     ---> System.InvalidOperationException: [Input] Pulumi.AzureNative.Network.Inputs.SubResourceArgs.Id must not be an Input<T>
       at Pulumi.InvokeArgs.ValidateMember(Type memberType, String fullName)
       at Pulumi.InputArgs..ctor()
       at Pulumi.InvokeArgs..ctor()
       at Pulumi.AzureNative.Network.Inputs.SubResourceArgs..ctor()
       at Systemorph.Cloud.Global.Network.ScNetworkGlobal..ctor() in .......\ScNetworkGlobal.cs:line 71
       --- End of inner exception stack trace ---
       at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
       at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
       at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
       at System.Activator.CreateInstance\[T\]()
       at Pulumi.Deployment.Runner.<>c__4\`1.<RunAsync>b__4_0()
       at Pulumi.Deployment.Runner.RunAsync[TStack](Func\`1 stackFactory)

The code in question is the following:

```
var virtualWan = new VirtualWan(Names.Network.Global.WanName, new VirtualWanArgs
{
	VirtualWANName = Names.Network.Global.WanName,
	Location = Names.Network.Global.AzureLocation,
	ResourceGroupName = wanRg.Name,
	Tags =
	{
		{ "Product", Names.General.ProductName },
		{ "Service", Names.General.Services.Network },
		{ "Region", Names.General.RegionNames.Global }
	},
	Type = "Basic",
	AllowVnetToVnetTraffic = true,
	DisableVpnEncryption = false
}, globalOptions);
       
var vhub = new VirtualHub(region.VHubName, new VirtualHubArgs // <--- this is line 71
{
	AddressPrefix = region.AddressPrefix,
	ResourceGroupName = rg.Name,
	Location = region.AzureLocation,
	Sku = "Basic",
	VirtualWan = new Pulumi.AzureNative.Network.Inputs.SubResourceArgs
	{
		Id = virtualWan.Id  // <---- here 
	},
	Tags =
	{
		{ "Product", Names.General.ProductName },
		{ "Service", Names.General.Services.Network },
		{ "Region", region.RegionName }
	},
	VirtualHubName = region.VHubName
}, options);
```

I'm using pulumi v2.25.2.

Posted question on Slack: https://pulumi-community.slack.com/archives/CRVK66N5U/p1618857718286400

**Comment by viveklak** [comment]
Fixed by https://github.com/pulumi/pulumi-azure-native/pull/769

**Comment by viveklak** [comment]
Will be available in 1.0.1
